### PR TITLE
do not use constrain_element_vector() with too many DOFs

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
OK, here's the problem --- `DofMap::constrain_element_XXX()` is designed to be used for numbers of DOFs that occur on individual elements, but I wrote `FEDataManager::interpWeighted()` to call it on, potentially, all of the DOFs in the mesh when using nodal interaction. Whoops.

**Comment by boyceg** [comment]
In particular, `DofMap::constrain_element_vector()` will create an n-by-n `DenseMatrix` in which n is set to the total number of DOF indices to check for constraints.

I think it should be fine to use `DofMap::enforce_constraints_exactly()` instead, and so that is what this PR does instead of calling `DofMap::constrain_element_vector()`.

The only thing I am not sure about is whether we can avoid calling `close()` on `F_vec`...

File: ibtk/src/lagrangian/FEDataManager.cpp [hunk 1]
```diff
@@ -1847,6 +1844,14 @@ FEDataManager::interpWeighted(const int f_data_idx,
     // Accumulate data.
     if (close_F) F_vec.close();
 
+    // Enforce constraints here if using nodal quadrature.  (If using
+    // element-based quadrature, constraints have already been imposed above.)
+    if (use_nodal_quadrature && F_dof_map.n_constrained_dofs() > 0)
+    {
+        if (!close_F) F_vec.close();
+        F_dof_map.enforce_constraints_exactly(F_system, &F_vec);
```
Comment (drwells) [inline ibtk/src/lagrangian/FEDataManager.cpp:1852]: This doesn't seem right - here F_vec is a RHS vector, not a solution vector. We should be performing assembly in the same way in both the nodal and nonnodal case.

**Comment by drwells** [comment]
The tests fail in the diff stage for me - there are substantial (6th decimal place) differences in the computed volumes.

File: ibtk/src/lagrangian/FEDataManager.cpp (hunk 1 above)
Comment (boyceg) [reply ibtk/src/lagrangian/FEDataManager.cpp:1852]: No --- nodal interpolation is just direct interpolation at the nodes --- it is nothing like the elemental case. We sort of fake it to be like the elemental case by multiplying by the diagonal mass matrix.

Comment (boyceg) [reply ibtk/src/lagrangian/FEDataManager.cpp:1852]: Maybe we need to impose the constraints on the assembled interpolated vector and then multiply by the diagonal mass matrix.

**Comment by boyceg** [comment]
> The tests fail in the diff stage for me - there are substantial (6th decimal place) differences in the computed volumes.

Because I cannot reliably run the test suite, I will need your help in fixing this. :-D

**Comment by drwells** [comment]
As discussed elsewhere: the root cause of the crash is the application of constraints when we use nodal quadrature. Since the degrees of freedom are uncoupled in this case we can completely skip any imposition of constraints until after the solution process, which is what this patch now does.

**Pull request merged**